### DISCUSSION TITLE
how does point tracker tracks IR blobs?

### DISCUSSION THREAD

**Discussion by mavizasyon** [opentrack/opentrack#1791] [Q&A] [answered]
I mean which alghorithm does it use? hough transformation?

**Comment by sthalik** [comment, marked as answer]
It uses meanshift to compute centroids. See: https://github.com/opentrack/opentrack/blob/master/tracker-pt/module/point_extractor.cpp#L52

**Reply by mavizasyon** [reply]
thank you for your answer
I think that algorithm got little bit problem,
![Do It Yourself _ Head Tracking _ TrackIR _ OpenTrack 32-42 screenshot](https://github.com/opentrack/opentrack/assets/51001692/ced530fe-25e8-4a4f-af22-b14e2eb690e0)
this is from a youtube video and in the image the algorithm combines two circles into one, isn't that a problem? I think it makes uncertinity, isn't hough transformation like in cv2 would be better?
sorry for my ignorance

**Reply by sthalik** [reply]
Play with the threshold inside tracker settings on the first tab. In automated mode, it's based on point sizes. Otherwise it's based on absolute brightness values. You may get better results by lowering the camera exposure and gain altogether.

You can try using the Hough circles algorithm but I doubt you'd get better results than properly configured thresholding using flood fill and meanshift as it is right now.
